### ISSUE TITLE
Read the extended component documentation from node_modules

### ISSUE THREAD

**Issue by nerijusb** [vue-styleguidist/vue-styleguidist#606]
Hi, awesome work on vue-styleguidist, keep up the good work! I have a small problem/possible improvement to discuss:

**The problem**

It comes from the restriction described in the comment from extendsHandler.ts:
`// only look for documentation in the current project not in node_modules`
This is problematic due to my specific setup:
I have an internal component library that is published as a separate npm package. In other project (that has vuestyleguidist) I use components from this library by extending them. When the documentation is generated for these new components, no parent documentation is picked up because the parent comes from module in node_modules.

**Proposed solution**

Introduce a new configuration option that would allow reading extended component information even if it comes node_modules.

**Comment by elevatebart** [comment]
Would you be open to creating a PR? I can guide you through the process.

**Comment by cbaxter713** [comment]
I am having the same issue.  Is there any update on progress here?

**Comment by elevatebart** [comment]
Hello @cbaxter713,

TLDR, it has not moved yet.

My concerns are:
1. parsers performance would probably suffer (a lot more files to run through)
1. Vsg users has no control over what other modules have documented so fixes would be hard
1. Modules that are compiled can be incompatible with Vsg.

We cannot make it work always, but an option is very much a good idea.

We could have a custom condition here, a predicate, instead of a hardcoded regular expression to bypass this limitation.
https://github.com/vue-styleguidist/vue-styleguidist/blob/dev/packages/vue-docgen-api/src/script-handlers/extendsHandler.ts#L40

Maybe a try-catch would help alleviate the incompatible components. It would be a silent failure though and would bring a lot of confusion.

What do you think?
Would you be open to experimenting with it in a PR?

**Comment by cbaxter713** [comment]
Thanks for the response @elevatebart .  I just want to confirm that we are talking about the same problem.  Here is an example of a component that I am documenting with VSG:

`
<script>
  import ZrButton from '@zaneray/vue-components/components/base/ZrButton.vue';

  /**
   * Button component that leverages the Zaneray BaseButton documented at https://zr-vue.netlify.com/#/Base%20Components/ZrButton
   */

  export default {
    name: "BaseButton",
    components: {
      ZrButton
    },
    props: {
      ...ZrButton.props,
      /**
       * Icon to display in the button.
       */
      icon: {
        type: String
      }
    }
  }
</script>
`

In the documentation for that component, the `icon` prop is documented but the `ZrButton.props` that get spread in are not documented.  Of course the props are working as expected when using the component, just not the documentation.  Can you confirm that is the same issue you were thinking?

**Comment by elevatebart** [comment]
It is a different issue. 

The issue @nerijusb is having is with the `extend` feature of node_modules.
Yours seem to be with the spreading of props.

Would you open another issue copying your use case?
Is the `@zaneray/vue-components/components/base/ZrButton.vue` an external package?

**Comment by cbaxter713** [comment]
Yes it is an external package.  I will create a new issue.  Thanks!

**Comment by elevatebart** [comment]
Hello @nerijusb 

I came up with a method that would I think help you a lot.
With the linked Pull Request you can now customize the function that validates if a component should be ignored when extending.
Would this would fit your need or do you see anything I miss?

Thank you for your request,
Bart
